google/cloud/spanner_dbapi/cursor.py

```diff
@@ -317,6 +317,10 @@ def executemany(self, operation, seq_of_params):
 
         many_result_set = StreamedManyResultSets()
 
+        # For every operation, we've got to ensure that any prior DDL
+        # statements were run.
+        self.connection.run_prior_DDL_statements()
```

Comment:
Yes what you said is true for `autocommit=false`. Here is the flow
- DDL statements as they come are batched
- Whenever an `execute` statement is called for DML/Query, first we run `run_prior_DDL_statements()` which execute all the batched DDL statements in the connection till now
- Then the actual DML/Query statement is executed
- In case of commit also we try to execute all the batched DDL statements

In case of `autocommit=true` DDL statements are executed as they come

Reply:
Thanks for clarifying